### ISSUE TITLE
Unexpected keyword argument in einsum call

### ISSUE THREAD

**Issue by stavros11** [nasa/hybridq#83]
Executing the following code on GPU using the jax backend:
```Python
from hybridq.circuit import Circuit
from hybridq.gate import Gate
from hybridq.circuit.simulation import simulate

circuit = Circuit()
circuit.append(Gate('RX', params=[0], qubits=(0,)))
circuit.append(Gate('RX', params=[0], qubits=(1,)))
circuit.append(Gate('RX', params=[0], qubits=(2,)))
circuit.append(Gate('H', qubits=(0,)))
circuit.append(Gate('H', qubits=(1,)))
circuit.append(Gate('H', qubits=(2,)))
circuit.append(Gate('CZ', qubits=(1, 2)))
circuit.append(Gate('RZ', params=[0], qubits=(0,)))
circuit.append(Gate('CZ', qubits=(0, 1)))
circuit.append(Gate('RX', params=[0], qubits=(0,)))
circuit.append(Gate('CZ', qubits=(1, 2)))
circuit.append(Gate('CZ', qubits=(1, 2)))
circuit.append(Gate('H', qubits=(1,)))
circuit.append(Gate('H', qubits=(2,)))

initial_state = len(circuit.all_qubits()) * '0'
simulate(circuit, optimize="evolution-einsum", backend="jax",
         initial_state=initial_state, compress=0, simplify=False)
```
leads to the following error:
```
TypeError: einsum() got an unexpected keyword argument 'dtype'
```
Removing any gate or changing any of the simulation options (eg. making `simplify=True` or `compress > 0`) fixes the issue and returns the correct state. For example the last two CZs cancel each other but if I remove them the error disappears. It is only the exact above configuration that gives me the error.

My GPU is a GTX 1650 with CUDA 11.2 and jax version 0.2.26 but I get the same issue in a different machine with CUDA 11.5. I installed hybridq using the
```
conda env create -f envinronment.yml
```
from the README.

**Comment by s-mandra** [comment]
Thanks for catching that! It seems that `opt_einsum.contract(backend='jax')` doesn't accept the keyword `dtype` (while `opt_einsum.contract(backend='numpy')` still does). Fixed in #85.

**Comment by stavros11** [comment]
I installed from source using the `fix-contract-einsum` branch and I can also confirm that it fixes the problem, thanks!
